object_manager.go

```diff
@@ -317,19 +336,51 @@ func (objMgr *ObjectManager) GetFixedAddress(netview string, cidr string, ipAddr
 	return &res[0], nil
 }
 
-func (objMgr *ObjectManager) UpdateFixedAddress(fixedAddrRef string, macAddress string, name string, vmID string) (*FixedAddress, error) {
+func (objMgr *ObjectManager) GetFixedAddressByRef(ref string) (*FixedAddress, error) {
+	fixedAddr := NewFixedAddress(FixedAddress{})
+	err := objMgr.connector.GetObject(fixedAddr, ref, &fixedAddr)
+	return fixedAddr, err
+}
 
+func (objMgr *ObjectManager) DeleteFixedAddress(ref string) (string, error) {
+	return objMgr.connector.DeleteObject(ref)
+}
+
+// validation  for match_client
+func validateMatchClient(value string) bool {
+	match_client := [5]string{"MAC_ADDRESS", "CLIENT_ID", "RESERVED", "CIRCUIT_ID", "REMOTE_ID"}
+
+	for _, val := range match_client {
+		if val == value {
+			return true
+		}
+	}
+	return false
+}
+
+func (objMgr *ObjectManager) UpdateFixedAddress(fixedAddrRef string, matchClient string, macAddress string, vmID string, vmName string) (*FixedAddress, error) {
 	updateFixedAddr := NewFixedAddress(FixedAddress{Ref: fixedAddrRef})
 
 	if len(macAddress) != 0 {
 		updateFixedAddr.Mac = macAddress
 	}
 
+	ea := objMgr.getBasicEA(true)
 	if vmID != "" {
-		ea := objMgr.getBasicEA(true)
 		ea["VM ID"] = vmID
 		updateFixedAddr.Ea = ea
 	}
+	if vmName != "" {
+		ea["VM Name"] = vmName
+		updateFixedAddr.Ea = ea
```

Comment:
I am not sure about this. So, you are basically updating the Fix Address's EA with Basic EA (plus these updates to "VM ID" and "VM Name" EAs). So, what if the Fix Address already have EAs, would they get overwritten? Have you tested it?

Reply:
yes it would get overwritten. vmId was already present , i did add vmName just to make sure attributes of vm would get covered

Reply:
What I mean is,  here `ea` only contain EAs returned by `getBasicEA()`, plus the two you set here. What if the Fixed Address you are updating has ***other*** EAs that are set by the user, for example, manually in NIOS. Would those EAs get erased?

Reply:
Nope ,they won't get erased.and that has been tested

Reply:
OK. Sounds good! Thanks!